core/src/main/java/com/netflix/conductor/core/execution/mapper/WaitTaskMapper.java

```diff
@@ -69,6 +79,52 @@ public List<TaskModel> getMappedTasks(TaskMapperContext taskMapperContext) {
         waitTask.setInputData(waitTaskInput);
         waitTask.setStartTime(System.currentTimeMillis());
         waitTask.setStatus(TaskModel.Status.IN_PROGRESS);
+        setCallbackAfter(waitTask);
         return List.of(waitTask);
     }
+
+    void setCallbackAfter(TaskModel task) {
+        String duration =
+                Optional.ofNullable(task.getInputData().get(DURATION_INPUT)).orElse("").toString();
+        String until =
+                Optional.ofNullable(task.getInputData().get(UNTIL_INPUT)).orElse("").toString();
+
+        if (StringUtils.isNotBlank(duration) && StringUtils.isNotBlank(until)) {
+            task.setReasonForIncompletion(
+                    "Both 'duration' and 'until' specified. Please provide only one input");
+            task.setStatus(FAILED_WITH_TERMINAL_ERROR);
+            return;
+        }
+
+        if (StringUtils.isNotBlank(duration)) {
+
+            Duration timeDuration = parseDuration(duration);
+            long waitTimeout = System.currentTimeMillis() + (timeDuration.getSeconds() * 1000);
+            task.setWaitTimeout(waitTimeout);
+            long seconds = timeDuration.getSeconds();
+            task.setCallbackAfterSeconds(seconds);
+
+        } else if (StringUtils.isNotBlank(until)) {
+            try {
+
+                Date expiryDate = parseDate(until);
+                long timeInMS = expiryDate.getTime();
+                long now = System.currentTimeMillis();
+                long seconds = ((timeInMS - now) / 1000) + 1;
```

Comment:
@manan164 Hello, I don't understand why we need to add 1 to the "seconds" variable. It seems to occasionally cause testWaitUntil to fail: https://github.com/Netflix/conductor/actions/runs/6294685428/job/17086871264?pr=3779

Reply:
Hi @Young-Zen , good point.  I think it may be for rounding purposes. I will change this. Thanks for pointing this out.

Reply:
Hi @Young-Zen , I have raised fix for this, https://github.com/Netflix/conductor/pull/3786